textlayout/resources/gap-analysis-github.js

```diff
@@ -1,218 +1,204 @@
 var owner = 'w3c';
 
-var sections = {}
-var debug = false
+var sections = {};
+var debug = false;
 
-var issues = []
-var maxpages = 5
+var issues = [];
+var maxpages = 5;
 
-var totals=0
-var counter=maxpages
-			
-function getAllData (repo, doc) {
-	for (var p=1;p<maxpages+1;p++) fetchIssues(p, repo)
-	var timer = setInterval(function() {
-		if (counter === 0) {
-			clearInterval(timer)
-			if (debug) console.log('finished','Issue length:',issues.length, sections)
-
-			// group issues by label, adding to the labels array
-			for (var i=0; i<issues.length; i++) {
-				if (issues[i].labels) {
-					for (var l=0;l<issues[i].labels.length;l++) { // for each label in that issue
-						if (issues[i].labels[l].name.startsWith('i:')) {
-							sLabelFound = issues[i].labels[l].name.replace(/^i:/,'')
-							if (sections[sLabelFound]) sections[sLabelFound].push(issues[i])
-							else {
-								sections[sLabelFound] = []
-								sections[sLabelFound].push(issues[i])
-								}
-							if (debug) console.log(sLabelFound)
-							}
-						}
-					}
-				}
-			buildDoc(repo, doc)
-			}
-		else if (debug) console.log(counter)
-		}, 50)
-	}
-
-	
-// Grab and present the issue list from GitHub
-function fetchIssues(page, repo) {
-	var request = new XMLHttpRequest();
-	request.open('get','https://api.github.com/repos/w3c/'+repo+'/issues?per_page=100&page='+page)
-	request.onload = function () {
-			var temp = JSON.parse(request.responseText)
-			for (var i=0;i<temp.length;i++) {
-				issues.push(temp[i])
-				}
-		totals += issues.length;
-		if (debug) console.log(issues.length,totals,page, counter)
-		counter--
-		}
-	request.send();
-	}
-				
+var totals=0;
+var counter=maxpages;
 
+function getAllData (repo, doc) {
+  let arr = [];
+  for (let p=1; p < maxpages+1; p++) {
+    arr.push(fetch('https://api.github.com/repos/w3c/'+repo+'/issues?per_page=100&page='+p));
```

Comment:
I suppose we need to check return code and throw custom error, that fetch does not throw error on 403/404 etc.
(also might be better to check maximum number of pages like #855 rather than using fixed number?)

Reply:
Indeed, it would be good to check the errors and go through all the pages. In that PR, I was just trying to fix the race condition issue for the snapshot generation.